tracer/test/Datadog.Trace.Tests/Configuration/TracerSettingsTests.cs

```diff
@@ -348,20 +348,23 @@ public void GlobalTags(string value, string legacyValue, string[] expected)
         [InlineData("", true, new string[0])]
         [InlineData("   ", true, new string[0])]
         // nominal
-        [InlineData("key1:value1,key2:value2,key3", true, new[] { "key1:value1", "key2:value2", "key3:" })]
+        [InlineData("key1:value1,key2:value2,key3", true, new[] { "key1|value1", "key2|value2", "key3|" })]
         // trim whitespace
-        [InlineData(" key1 : value1 ", true, new[] { "key1:value1" })]
+        [InlineData(" key1 : value1 ", true, new[] { "key1|value1" })]
         // other normalization
-        [InlineData("key1:val.u e1?!", true, new[] { "key1:val.u e1?!" })]
-        [InlineData("k.e y?!1", false, new[] { "k.e y?!1:" })]
+        [InlineData("key1:val.u e1?!", true, new[] { "key1|val.u e1?!" })]
+        [InlineData("k.e y?!1", false, new[] { "k.e y?!1|" })]
+        [InlineData(":leadingcolon", false, new string[0])]
+        [InlineData(":leadingcolon", true, new string[0])]
+        [InlineData("one:two:three", true, new[] { "one:two|three" })]
```

Comment:
Should we have a case for `"one:two:three", false`? Are these values from the system tests? Might be nice to include those here too if not

Reply:
> Should we have a case for "one:two:three", false?

No point, it doesn't change anything. The "header fix" is not related to the behaviour in this PR, it's about how invalid characters are normalized.